packages/admin/src/pages/super-admin-dashboard/user/[id]/change-roles.page.tsx

```diff
@@ -44,17 +54,22 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
     userDepartment,
     newUserRoles,
     userId,
+    departmentPageUrl,
   }: Awaited<ReturnType<typeof handleRequest>>) {
-    const ADMIN_ROLES_IDS = ['3', '4', '5'];
-
     const userHasDepartment = userDepartment !== null;
-    const userBecomingApplicant = !newUserRoles.some((role) =>
-      ADMIN_ROLES_IDS.includes(role)
-    );
+    const userBecomingApplicant = !newRolesAreAdminRoles(newUserRoles);
 
     return userHasDepartment || userBecomingApplicant
       ? `/super-admin-dashboard/user/${userId}`
-      : `/super-admin-dashboard/user/${userId}/change-department`;
+      : departmentPageUrl;
+  }
+
+  function newRolesAreAdminRoles(newRoles: string[]) {
+    return newRoles.some((role) => ADMIN_ROLES_IDS.includes(role));
+  }
+
+  function isAlreadyAdmin(oldRoles: string[]) {
+    return oldRoles.some((role) => ADMIN_ROLES_IDS.includes(role));
   }
```

Comment:
These two functions do the same thing, so you can just have the one function with a more generic name:

```suggestion
  function hasAdminRole(roles: string[]) {
    return roles.some((role) => ADMIN_ROLES_IDS.includes(role));
  }
```